examples/erc20/tests/erc20.spec.ts

```diff
@@ -1,94 +1,74 @@
-    this.timeout(150000);
-    ({
-      contract: token,
-      payerETHAddress,
-      contractAbi,
-      program,
-      contractStorageKeyPair: tokenStorageKeyPair,
-    } = await loadContract(__dirname, [NAME, SYMBOL, TOTAL_SUPPLY]));
-
-    const { result: name } = await token.functions.name();
-    expect(name).toEqual('Solana');
-
-    const { result: symbol } = await token.functions.symbol();
-    expect(symbol).toEqual('SOL');
-
-    const { result: decimals } = await token.functions.decimals();
-    expect(decimals).toEqual(18);
-
-    const { result: supply } = await token.functions.totalSupply();
-    expect(supply.toString()).toEqual(TOTAL_SUPPLY.toString());
-
-    const { result: balance } = await token.functions.balanceOf(
-      payerETHAddress
-    );
-    expect(balance.toString()).toEqual(TOTAL_SUPPLY.toString());
-  });
-
-  it('loads existing contract', async function () {
-    token = await program.getContract(
-      contractAbi,
-      tokenStorageKeyPair.publicKey
-    );
-
-    const { result: name } = await token.functions.name();
-    expect(name).toEqual('Solana');
-  });
-
-  it('mutates contract state', async function () {
-    const otherAccount = pubKeyToHex(Keypair.generate().publicKey);
-    const transferAmount = 9;
-
-    await token.functions.transfer(otherAccount, transferAmount);
-
-    const { result: otherAccountBalance } = await token.functions.balanceOf(
-      otherAccount
-    );
-    expect(otherAccountBalance.toString()).toEqual(transferAmount.toString());
-
-    const { result: payerBalance } = await token.functions.balanceOf(
-      payerETHAddress
-    );
-    expect(payerBalance.toString()).toEqual(
-      (TOTAL_SUPPLY - transferAmount).toString()
-    );
-  });
-
-  it('emits events', async function () {
-    const spenderAccount = pubKeyToHex(Keypair.generate().publicKey);
-    const spendAmount = 9;
-
-    const event: LogDescription = await new Promise((resolve) => {
-      let listenId = token.addEventListener(async (event) => {
-        await token.removeEventListener(listenId);
-        resolve(event);
-      });
-      token.functions.approve(spenderAccount, spendAmount);
+    let contract: Contract;
+    let payerETHAddress: string;
+
+    it('deploys new contract', async function () {
+        this.timeout(150000);
+        ({ contract, payerETHAddress } = await loadContract(__dirname, [NAME, SYMBOL, TOTAL_SUPPLY]));
+
+        const name = await contract.name();
+        expect(name).toEqual(NAME);
+
+        const symbol = await contract.symbol();
+        expect(symbol).toEqual(SYMBOL);
+
+        const decimals = await contract.decimals();
+        expect(decimals).toEqual(18);
+
+        const supply = await contract.totalSupply();
+        expect(supply.toString()).toEqual(TOTAL_SUPPLY.toString());
+
+        const balance = await contract.balanceOf(payerETHAddress);
+        expect(balance.toString()).toEqual(TOTAL_SUPPLY.toString());
```

Comment:
`contract.functions.name(...)` can still be used, and when called returns the full `{ results, logs, events, computeUnitsUsed }` object.

When calling `contract.name(...)`, only the result is returned unwrapped. This roughly mirrors the behavior of Ethers.js.